src/test/shell/bazel/external_integration_test.sh

```diff
@@ -925,6 +925,22 @@ EOF
 }
 
 
+function test_server_timeout() {
+  startup_timeout_server $PWD
+
+  cat > WORKSPACE <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "repo",
+    url = "http://127.0.0.1:$fileserver_port/some_archive.zip",
+)
+EOF
+  bazel build @repo//... &> $TEST_log 2>&1 && fail "Expected to fail"
```

Comment:
How long does it wait for a timeout? I'm wondering how bad the effect on our CI is going to be...

Reply:
This test takes ~3 mins.

I did ```time bazel build @repo//... &> $TEST_log 2>&1 && fail "Expected to fail"``` 
in line 938 and got:
```
** test_server_timeout *********************************************************

real    2m56.609s
user    0m0.020s
sys     0m0.016s
PASSED: test_server_timeout
```

Reply:
I think that's a problem. Having a test wait for 3 minutes doing nothing means our CI will take 3 minutes longer. Is there any way to change the timeout with a flag or something like that?

Reply:
> Is there any way to change the timeout with a flag or something like that?

Not at the moment. All the timeouts are hard-coded. But adding a flag to configure those timeouts seems a good idea in general; will work on this.